TopsortAnalytics/src/main/java/com/topsort/analytics/model/ClickEvent.kt

```diff
@@ -41,7 +42,32 @@ data class Click(
      * The marketplace's ID for the click
      */
     val id: String,
-)
+) {
+    fun toJsonObject(): JSONObject {
+        return JSONObject()
+            .put("resolvedBidId", resolvedBidId)
+            .put("entity", entity?.toJsonObject())
+            .put("additionalAttribution", additionalAttribution)
+            .put("placement", placement.toJsonObject())
+            .put("occurredAt", occurredAt)
+            .put("opaqueUserId", opaqueUserId)
+            .put("id", id)
```

Comment:
With this you may end up with an object that has both a `resolvedBidId` and an `entity` which is a bad request according to our API.
The best solution is to ask if whether it has a `resolvedBidId`, then use that, otherwise, use the entity.

Reply:
Changed it, also changed the constructor to private and added Factories to Click and Impressions to discipline these objects creation.